source/gen_security.rst

```diff
@@ -1,37 +1,371 @@
 
-Any service that supports the OAuth2 protocol can be used to provide authentication and authorization to a Odahu cluster.
+2. HTTP Request from inside the cluster perimeter. Such requests usually made by background processes inside the cluster
+   on behalf of service accounts.
 
-.. warning::
+   1. Service should previously authenticate in idP using `OpenID Connect`_ protocol.
+      The most suitable way to authenticate services is `OAuth2 Client Credentials Grant`_
+   2. Service makes a request to API using issued JWT token as `Authorization Request Header Field`_
 
-    Connectivity from cluster to OAuth2 IP (identity provider) is required.
+3. Envoy proxy as PEP_ that is configured as sidecar container by `Istio Pilot`_ for those ODAHU components that must be
+   protected ensures that security policy allows making this request to ODAHU API
 
-Odahu integrations OAuth2 to Identity Providers using the ``oauth2_proxy`` library.
+   1. Envoy verifies JWT token in `Authorization Request Header Field`_  using `JSON Web Token (JWT) Authentication`_ filter
+   2. Envoy makes a query to OpenPolicyAgent_ as PDP_ using `External Authorization`_ filter passing parsed JWT token from the previous step
 
-Securing model APIs
--------------------
+4. If request if permitted by OpenPolicyAgent_ it is sent to upstream (ODAHU API)
 
-:term:`Model prediction API` and :term:`Model Feedback API` calls are secured using :term:`JWT tokens <JWT Token>`.
-:term:`API service` can issue a new :term:`JWT token` for a given :term:`role name <Model Deployment Access Role Name>`.
-:term:`JWT tokens <JWT Token>` are scoped to specific :term:`role names <Model Deployment Access Role Name>` and may be used for granting access to the :term:`Model prediction API`.
-The role name has to be defined during creation of :term:`model deployments <Deploy>`, otherwise a hard-coded default value will be used.
 
-Securing management APIs
+***********************
+Use built-in policies
+***********************
+
+ODAHU is distributed with build-in policies that are written on `Rego policy language`_.
+
+`Role based access control`_ is implemented by default.
+But there is nothing you need to know about rego policies language until
+the set of pre-defined roles fits your needs. In this section, you can see such roles and how to manage them
+using Keycloak as an example of idP_
+
+Pre-defined roles overview
+============================
+
+ODAHU has next predefined roles:
+
+  #. 'odahu_viewer' – can do any GET requests to any API resources
+  #. 'odahu_flow_scientist' – allowed requests:
+
+     #. Action: GET, Resource pattern: api/v1/connection.*
+     #. Action: GET, Resource pattern: api/v1/packaging/integration.*
+     #. Action: GET, Resource pattern: api/v1/toolchain/integration.*
+     #. Action pattern: .*, Resource:  api/v1/model/training.*
+     #. Action pattern: .*, Resource:  api/v1/model/packaging.*
+     #. Action pattern: .*, Resource:  api/v1/model/deployment.*
```

Comment:
Is it possible to change '.*' on something more clear for persons of the business domain?
e. g.  Action: `ALL` , Resource pattern: `api/v1/packaging/training` and stop there.